lightning-invoice/src/utils.rs

```diff
@@ -229,7 +231,7 @@ where
 ///
 /// [`PhantomKeysManager`]: lightning::sign::PhantomKeysManager
 fn select_phantom_hints<L: Deref>(amt_msat: Option<u64>, phantom_route_hints: Vec<PhantomRouteHints>,
-	logger: L) -> Vec<RouteHint>
+	logger: L) -> impl std::iter::Iterator<Item = RouteHint>
```

Comment:
Since you import `Iterator`, then you don't need to fully qualify it here.

Reply:
Good catch.  I didn't clean that up after adding the import statement.